dali/kernels/audio/mel_scale/mel_filter_bank_gpu.cu

```diff
@@ -16,28 +16,61 @@
 #include <memory>
 #include "dali/kernels/audio/mel_scale/mel_filter_bank_gpu.h"
 #include "dali/core/tensor_shape_print.h"
+#include "dali/core/cuda_rt_utils.h"
+#include "dali/core/static_switch.h"
+#include "dali/core/fast_div.h"
 
 namespace dali {
 namespace kernels {
 namespace audio {
 
 const int kBlockDim2 = 32;
-const int kBlockDim1 = 1024;
 
 template <typename T>
-struct BlockDesc {
-  const T *in_frame;
-  T *out_frame;
-  union {
-    struct {  // outer-dim fft
-      int64_t start_window;
-      int64_t frame_nwindows;
-    };
-    struct {  // inner-dim fft
-      int64_t block_start;
-      int64_t out_frame_size;
-    };
-  };
+struct BlockDescOuter {
+  BlockDescOuter() = default;
+  BlockDescOuter(int64_t start_window, int64_t nwindows) {
+    this->out_frame  = nullptr;
+    this->in_frame   = nullptr;
+
+    this->start_window = start_window;
+    this->frame_nwindows = nwindows;
+  }
+
+  void SetBasePointers(T *out, const T *in) {
+    out_frame = out;
+    in_frame  = in;
+  }
+
+  T        *out_frame;
+  const T  *in_frame;
+
+  int64_t start_window;
+  int64_t frame_nwindows;
+};
+
+template <typename T>
+struct BlockDescInner {
+  BlockDescInner() = default;
+
+  BlockDescInner(int start_window, int end_window) {
+    this->out_frame  = nullptr;
+    this->in_frame   = nullptr;
+
+    this->start_window = start_window;
+    this->end_window = end_window;
+  }
+
+  void SetBasePointers(T *out, const T *in) {
+    out_frame = out;
+    in_frame  = in;
+  }
+
+  T        *out_frame;
+  const T  *in_frame;
+
+  int start_window;
+  int end_window;
```

Comment:
Switched to int - with 1ms hop it can handle a 24-day long recording. I guess that's enough.